lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_method.rb

```diff
@@ -24,9 +24,17 @@ def self.invoke_ansibletemplate(aem, obj, inputs)
     end
 
     def self.invoke_uri(aem, obj, _inputs)
-      scheme, userinfo, host, port, registry, path, opaque, query, fragment = URI.split(aem.data)
-      raise  MiqAeException::MethodNotFound, "Specified URI [#{aem.data}] in Method [#{aem.name}] has unsupported scheme of #{scheme}; supported scheme is file" unless scheme.downcase == "file"
-      raise  MiqAeException::MethodNotFound, "Invalid file specification -- #{aem.data}" if path.nil?
+      uri = URI.parse(aem.data)
+
+      unless uri.scheme.casecmp?("file")
+        msg = "Specified URI [#{aem.data}] in Method [#{aem.name}] has unsupported scheme of #{scheme}; supported scheme is file"
+        raise MiqAeException::MethodNotFound, msg
+      end
+
+      if uri.path.blank?
+        raise  MiqAeException::MethodNotFound, "Invalid file specification -- #{aem.data}"
+      end
+
       # Create the filename corresponding to the URI specification
       fname = ae_methods_dir.join(path)
```

Comment:
@djberg96 should this path be changed to be uri.path

Reply:
Yep, good catch, updated.